torchaudio/datasets/tedlium.py

```diff
@@ -0,0 +1,158 @@
+
+        Raises:
+            RuntimeError: If release identifier does not match any supported release,
+        """
+        self._ext_audio = audio_ext
+        if release in _RELEASE_CONFIGS.keys():
+            folder_in_archive = _RELEASE_CONFIGS[release]["folder_in_archive"]
+            url = _RELEASE_CONFIGS[release]["url"]
+            subset = subset if subset else _RELEASE_CONFIGS[release]["subset"]
+        else:
+            # Raise warning
+            raise RuntimeError(
+                "The release {} does not match any of the supported tedlium releases{} ".format(
+                    release, _RELEASE_CONFIGS.keys(),
+                )
+            )
+
+        basename = os.path.basename(url)
+        archive = os.path.join(root, basename)
+
+        basename = basename.split(".")[0]
+
+        self._path = os.path.join(root, folder_in_archive, _RELEASE_CONFIGS[release]["data_path"])
+        if subset in ["train", "dev", "test"]:
+            self._path = os.path.join(self._path, subset)
+        if download:
+            if not os.path.isdir(self._path):
+                if not os.path.isfile(archive):
+                    checksum = _RELEASE_CONFIGS[release]["checksum"]
+                    download_url(url, root, hash_value=checksum)
+                extract_archive(archive)
+
+        walker = walk_files(self._path, suffix=".stm", prefix=False, remove_suffix=True)
+        self._walker = list(walker)
+        self._extended_walker = []
+        for file in self._walker:
+            stm_path = os.path.join(self._path, "stm", file + ".stm")
+            with open(stm_path) as f:
+                l = len(f.readlines())
+                self._extended_walker += [(file, line) for line in range(l)]
+
+    def load_tedlium_item(self, fileid: str, line: int, path: str) -> Tedlium_item:
+        """Loads a TEDLIUM dataset sample given a file name and corresponding sentence name
+
+        Args:
+            fileid (str): File id to identify both text and audio files corresponding to the sample
+            line (int): Line identifier for the sample inside the text file
+            path (str): Dataset root path
+
+        Returns:
+            Tedlium_item: A namedTuple containing [waveform, sample_rate, transcript, talk_id, speaker_id, identifier]
+        """
+        transcript_path = os.path.join(path, "stm", fileid)
+        with open(transcript_path + ".stm") as f:
+            transcript = f.readlines()[line]
+            talk_id, _, speaker_id, start_time, end_time, identifier, transcript = transcript.split(" ", 6)
+
+        wave_path = os.path.join(path, "sph", fileid)
+        waveform, sample_rate = self.load_audio(wave_path + self._ext_audio)
+        # Calculate indexes for start time and endtime
+        start_time = int(float(start_time) * sample_rate)
+        end_time = int(float(end_time) * sample_rate)
+        waveform = waveform[:, start_time:end_time]
```

Comment:
You can use `frame_offset`, and `num_frames` arguments to perform slicing directory.
This will not load the entire audio file, so it will be faster.

To do this, you need to know `sample_rate` beforehand.
You can either hardcode 16000 or do `torchaudio.info` to get the sample rate dynamically.
I think `torchaudio.info` can incur performance issue because it will open each file twice.
Hardcoding 16000 should be fine because all the TED-LIUM dataset is 16k[Hz].

Reply:
I think we should go with a hardcoded sample rate then, will update in a following commit

Reply:
I found some inconsistency between sox and sox_io load functions:

https://github.com/pytorch/audio/blob/c692fe99ac353f0e2fc7b5ad7162f037a70a35fe/torchaudio/backend/sox_backend.py#L20-L28

https://github.com/pytorch/audio/blob/c692fe99ac353f0e2fc7b5ad7162f037a70a35fe/torchaudio/backend/sox_io_backend.py#L37-L43

`offset` vs `frame_offset`

Is this intended? This will cause some trouble when changing the backend if we make use of the parameter here.

Reply:
This is intended. "sox" backend is going to be deprecated in the next release and "sox_io" backend will become the default backend, then "sox" backend will be removed.